--registry parameter is not effective,when `.modules.yaml` exist

### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

_No response_

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI, Lockfile, Package manager compatibility

### Link to the code that reproduces this issue or a replay of the bug

_No response_

### Reproduction steps

```
// success
pnpm i lodash
//error
pnpm install my-private-repo   --registry=http://127.0.0.1:5555  -D
```
throw error
```
 ERR_PNPM_FETCH_404  GET https://registry.npmjs.org/my-private-repo: Not Found - 404

This error happened while installing a direct dependency of /home/user/my-project/adfs

my-private-repo is not in the npm registry, or you have no permission to fetch it.

An authorization header was used: Bearer npm_[hidden]
Progress: resolved 1, reused 1, downloaded 0, added 0
```
if I install `my-private-repo` first(without node_modules exist),It will install success


### Describe the Bug

I think `node_modules/.modules.yaml` file save `registries` and then `--registry` not work

### Expected Behavior

This operation can be successfully executed on `NPM`, so I hope it can also be compatible



### Which Node.js version are you using?

20.10.0

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [X] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

Ubuntu 22

--- a/.changeset/funny-snails-turn.md
+++ b/.changeset/funny-snails-turn.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/get-context": patch
+"pnpm": patch
+---
+
+Registry configuration from previous installation should not override current settings [#7507](https://github.com/pnpm/pnpm/issues/7507).

--- a/pkg-manager/get-context/src/index.ts
+++ b/pkg-manager/get-context/src/index.ts
@@ -172,10 +172,7 @@ export async function getContext (
     pendingBuilds: importersContext.pendingBuilds,
     projects: Object.fromEntries(importersContext.projects.map((project) => [project.rootDir, project])),
     publicHoistPattern: importersContext.currentPublicHoistPattern ?? opts.publicHoistPattern,
-    registries: {
-      ...opts.registries,
-      ...importersContext.registries,
-    },
+    registries: opts.registries,
     rootModulesDir: importersContext.rootModulesDir,
     skipped: importersContext.skipped,
     storeDir: opts.storeDir,

